src/metaswitch/clearwater/etcd_tests/etcdtestbase.py

```diff
@@ -17,7 +17,13 @@ def test_basic_clustering(self):
         c = EtcdCluster(2)
         s1, s2 = c.servers.values()
 
-        hasOneLeader = s1.isLeader() != s2.isLeader()
+        s1_leader = s1.isLeader()
+        s2_leader = s2.isLeader()
+        hasOneLeader = s1_leader != s2_leader
+        if not hasOneLeader:
+            print ("\ns1: {}, isLeader {}"
```

Comment:
Instead of doing this, another option would be add methods to EtcdCluster and EtcdServer which write out their state and then do something like

`print "\n Cluster: " + c.debug()`

I'd have the debug methods note everything interesting about the object - e.g. including logging the state directly - rather than the current test at https://github.com/Metaswitch/clearwater-etcd/blob/dev/src/metaswitch/clearwater/etcd_tests/etcdserver.py#L129

Reply:
Note, I'd do this rather doing this on `__repr__` as `__repr__` probably shouldn't call out to a HTTP API. You may also want to log the check, so maybe:

```
print "\n Should have one leader: {!r} - {}, {!r} - {}", s1, s1_leader, s2, s2_leader"
print "\n Cluster: " + c.debug()
```

and add a repr to EtcdServer which just logs the static data including IP.